sts/signer.go

```diff
@@ -274,11 +274,20 @@ func (s *Signer) SignRequest(req *http.Request) error {
 		}
 
 		var buf bytes.Buffer
+		host := req.URL.Hostname()
+		// Check if the host IP is in IPv6 format. If yes, add the opening and closing square brackets.
+		// If the host IP contains ":", it is either an IPv6 address or the entire host-port input if the port is not valid.
+		// https://github.com/golang/go/blob/master/src/net/url/url.go#L1148
+		// So we should rule out the possibility when IPv4:port is returned.
+		if strings.Contains(host, ":") && !strings.Contains(host, ".") {
+			host = fmt.Sprintf("%s%s%s", "[", host, "]")
+		}
```

Comment:
Have some concerns here.
- need to check if this is the right way to distinguish ip format. Per discussion in https://stackoverflow.com/questions/22751035/golang-distinguish-ipv4-ipv6, seems we could use the `ParseIP` method from `net` package
- not a blocker, should we add checkers say `IsIPv4` and `IsIPv6`, and invoke in `SignRequest`, considering they would be reused somewhere else

Reply:
Good point. I used `govalidator` instead, which has IsIPv6 check.